A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js

```diff
@@ -1275,7 +1275,19 @@ class FMCMainDisplay extends BaseAirliners {
     }
 
     checkVSpeedDisagree(mcdu) {
-        return mcdu.v1Speed && mcdu.vRSpeed && mcdu.v2Speed && mcdu.v1Speed <= mcdu.vRSpeed && mcdu.vRSpeed <= mcdu.v2Speed;
+        return (!!mcdu.v1Speed && !!mcdu.vRSpeed ? mcdu.v1Speed <= mcdu.vRSpeed : true)
+            && (!!mcdu.vRSpeed && !!mcdu.v2Speed ? mcdu.vRSpeed <= mcdu.v2Speed : true)
+            && (!!mcdu.v1Speed && !!mcdu.v2Speed ? mcdu.v1Speed <= mcdu.v2Speed : true);
+    }
+
+    vSpeedDisagreeCheck(
+        v1 = SimVar.GetSimVarValue("L:AIRLINER_V1_SPEED", "Knots"),
+        vr = SimVar.GetSimVarValue("L:AIRLINER_VR_SPEED", "Knots"),
+        v2 = SimVar.GetSimVarValue("L:AIRLINER_V2_SPEED", "Knots")
+    ) {
+        if ((!!v1 && !!vr ? v1 > vr : false) || (!!vr && !!v2 ? vr > v2 : false) || (!!v1 && !!v2 ? v1 > v2 : false)) {
+            this.addNewMessage(NXSystemMessages.vToDisagree, this.checkVSpeedDisagree);
+        }
```

Comment:
There's quite some duplicate logic here. It's important to note that the second argument to `addNewMessage` is not mandatory. There are other usages with type II messages and I assume those also work. Therefore, you might consider changing this code to:

```javascript
    checkVSpeedDisagree(v1, vr, v2) {       
        return (!!v1 && !!vr ? v1 <= vr : true)
            && (!!vr && !!v2 ? vr <= v2 : true)
            && (!!v1 && !!v2 ? v1 <= v2 : true);
    }

    vSpeedDisagreeCheck(
        v1 = SimVar.GetSimVarValue("L:AIRLINER_V1_SPEED", "Knots"),
        vr = SimVar.GetSimVarValue("L:AIRLINER_VR_SPEED", "Knots"),
        v2 = SimVar.GetSimVarValue("L:AIRLINER_V2_SPEED", "Knots")
    ) {
        if (this.checkVSpeedDisagree(v1, vr, v2)) {
            this.addNewMessage(NXSystemMessages.vToDisagree);
        }
   }
```

Note that `undefined && 123 ? true : false` results in `false`, meaning the `!!` in `checkVSpeedDisagree` isn't necessary. Perhaps you aimed to determine if values were 0? `0 && 1` evaluates to `0` which evaluates to `false`. Therefore, when `0` is entered as e.g. v1 it will be accepted. I am aware that the `trySetxxSpeed` functions check the range, however it is probably best to not assume so within this function

Reply:
with this changes checkVSpeedDisagree can no longer get called from the messages verifier that the message is still valid.

Reply:
Ok. That's good to know. Indeed I hadn't figured out yet that it verifies messages are still valid at a later time 👍 .

Then perhaps this is still an option?

```javascript
    checkVSpeedDisagree() {       
        return (!!this.v1Speed && !!this.vRSpeed ? this.v1Speed <= this.vRSpeed : true)
            && (!!this.vRSpeed && !!this.v2Speed ? this.vRSpeed <= this.v2Speed : true)
            && (!!this.v1Speed && !!this.v2Speed ? this.v1Speed <= this.v2Speed : true);
    }

    vSpeedDisagreeCheck() {
        if (this.checkVSpeedDisagree()) {
            this.addNewMessage(NXSystemMessages.vToDisagree, this.checkVSpeedDisagree.bind(this));
        }
    }
```

Reply:
That would work, and I could've made it that way, but I wanted to keep the original variables in place. (the sim vars) because I am not whether they were necessary or not.